chain/store.go

```diff
@@ -30,19 +30,30 @@ var logStore = logging.Logger("chain.store")
 
 var headKey = datastore.NewKey("/chain/heaviestTipSet")
 
+type globalIPLDSource struct {
+	// bs is the blockstore from which IPLD objects are pulled
+	bs bstore.Blockstore
+}
+
+// GetBlock retrieves a filecoin block by cid from the global IPLD source.
+func (source *globalIPLDSource) GetBlock(ctx context.Context, c cid.Cid) (*types.Block, error) {
```

Comment:
Initially the addition of this method and structure seemed out of place to me. I _think_ we are adding this for `LoadTipSetBlocks` and thus the `TipsetIterator` to use, is that correct? I don't really like this, but I don't have a better suggestion. Gut feeling says `IterAncestors` could be a method on the ChainStore, but maybe there is a better way. 

Fine leaving this the way it is -- it's an improvement, but I think we can make this cleaner after a few more related changes land.

Reply:
>  I think we are adding this for LoadTipSetBlocks and thus the TipsetIterator to use, is that correct?

correct
>  I don't really like this

me neither

> I think we can make this cleaner after a few more related changes land.

It seems like we're having the same thoughts. Feels good!

Reply:
This makes sense to me as as you have it, so far.